src/utils.jl

```diff
@@ -15,22 +15,6 @@ function use_testpattern()
     testpattern()
 end
 
-# --------------------------------------------------------------------
-# rand() is not threadsafe (https://discourse.julialang.org/t/4683)
-
-# Because we only require random numbers to sample parameters
-# and not the actual expensive computation, this seems like a better
-# approach than using separate RNG per thread.
-const rand_mutex = Ref{Threads.Mutex}()
-
-# constant overhead of about 80 ns compared to unsafe rand
-function safe_rand(args...)
```

Comment:
I'm a little worried here that I might misunderstand what "threadsafe" means here. Is that mean the following two scripts should output the same results?

```julia
Random.seed!(0)
Threads.@threads for i in 1:5
    println(rand())
end
```

```julia
Random.seed!(0)
for i in 1:5
    println(rand())
end
```

Reply:
no. This has no influence on the ordering that threads invoke `save_rand`. It really just is about the function being threadsafe so that consecutive return values (regardless of calling thread) can be considered random